default/python/universe_generation/universe_generator.py

```diff
@@ -146,7 +147,7 @@ def create_universe(psd_map):
                 report_error("Python create_universe: couldn't set up empire for team %d" % team)
             psds = psds_new
         # place leftovers
-        for (empire, psd), home_system in zip(psds, home_systems):
+        for (empire, psd), home_system in zip(psds, hs):
```

Comment:
Um, maybe I'm confused or missing something, but above (lines 134-136) you make a copy of the original `home_systems` list and assign it to `hs`. Then you use the copy in `hs` to work with in the subsequent code, and remove the home systems processed from `hs`.

If you then use `hs` (which should now have the home systems removed) instead of the original `home_systems` (which should be unaltered, so still containing the entire list of home systems), then nothing has changed, the bug remains.

Shouldn't `home_systems` be used here to iterate over?

Reply:
No, `home_systems` is a list of all home systems and `hs` contains not yet processed home systems for team placement. If there some players without team assigned they should be processed here.

Reply:
Ah, I see. Yep, my mistake 😉